orquesta/expressions/jinja.py

```diff
@@ -53,16 +54,23 @@ class JinjaEvaluator(expr_base.Evaluator):
     _regex_pattern = '{{.*?}}'
     _regex_parser = re.compile(_regex_pattern)
 
-    _regex_dot_pattern = '[a-zA-Z0-9_\'"\.\[\]\(\)]*'
-    _regex_ctx_pattern_1 = 'ctx\(\)\.%s' % _regex_dot_pattern
-    _regex_ctx_pattern_2 = 'ctx\([\'|"]?{0}[\'|"]?\)[\.{0}]?'.format(_regex_dot_pattern)
-    _regex_var_pattern = '.*?(%s|%s).*?' % (_regex_ctx_pattern_1, _regex_ctx_pattern_2)
-    _regex_var_parser = re.compile(_regex_var_pattern)
-
-    _regex_dot_extract = '([a-zA-Z0-9_\-]*)'
-    _regex_ctx_extract_1 = 'ctx\(\)\.%s' % _regex_dot_extract
-    _regex_ctx_extract_2 = 'ctx\([\'|"]?%s(%s)' % (_regex_dot_extract, _regex_dot_pattern)
-    _regex_var_extracts = ['%s\.?' % _regex_ctx_extract_1, '%s\.?' % _regex_ctx_extract_2]
+    _regex_ctx_pattern = '[a-zA-Z0-9_\'"\.\[\]\(\)]*'
+    _regex_ctx_patterns = [
+        '^ctx\(\)\.%s' % _regex_ctx_pattern,                                # line start ctx().*
+        '^ctx\([\'|"]?{0}[\'|"]?\)[\.{0}]?'.format(_regex_ctx_pattern),     # line start ctx(*).*
```

Comment:
This regex is a tiny bit too broad.

1. It allows `ctx(|asdf|)` to be a valid reference, which I don't think is the intent.
2. It allows `ctx(|asdf')` to be a valid reference.

I would recommend this instead:

```
'^ctx\(('{0}'|"{0}"|{0})\)[\.{0}]?'
```

Reply:
This is ok. If you go thru the var extraction logic, there are two stages.  The first stage is to extract any `ctx().*` and `ctx(*)` so it is ok to be as board as possible.  The second stage then takes those substrings from the first stage and then pull out the actual variables.  I've added more unit tests to ensure I don't extract variables like `ctx("foo')`, `ctx(foo")`, or etc. The reason for the two stages is to avoid regex pattern too complex to troubleshoot/debug.